From f7702086585465f6ccaa33ae815535e8ff10f025 Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Fri, 23 Sep 2011 11:05:04 +0800
Subject: [PATCH] ENGR00157473 MX5X: UART: disable UART2 DMA to make GPS work.

After enable DMA, GPS will keep report these DMA error:

   UART: DMA_ERROR: sr1:2010 sr2:508b
   UART: DMA_ERROR: sr1:2050 sr2:508a
   UART: DMA_ERROR: sr1:2050 sr2:508b
   UART: DMA_ERROR: sr1:10 sr2:1083
   UART: DMA_ERROR: sr1:50 sr2:1082
   UART: DMA_ERROR: sr1:2010 sr2:508b
   UART: DMA_ERROR: sr1:2050 sr2:508a
   UART: DMA_ERROR: sr1:2010 sr2:508b
   UART: DMA_ERROR: sr1:2010 sr2:508b
   UART: DMA_ERROR: sr1:2010 sr2:508b
   UART: DMA_ERROR: sr1:50 sr2:1083

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
(cherry picked from commit ddaf091fd3f5fae56b3c83f5cf59ee4f189f0a40)
---
 arch/arm/mach-mx5/serial.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-mx5/serial.h b/arch/arm/mach-mx5/serial.h
index 6ed55da..b142fdb 100644
--- a/arch/arm/mach-mx5/serial.h
+++ b/arch/arm/mach-mx5/serial.h
@@ -45,7 +45,7 @@
 #define UART1_DMA_ENABLE	0
 /* UART 2 configuration */
 #define UART2_UCR4_CTSTL        -1
-#define UART2_DMA_ENABLE	1
+#define UART2_DMA_ENABLE	0
 #define UART2_DMA_RXBUFSIZE     512
 #define UART2_UFCR_RXTL         16
 #define UART2_UFCR_TXTL         16
-- 
1.5.4.4

